app/models.py

```diff
@@ -0,0 +1,72 @@
+from werkzeug.security import generate_password_hash, check_password_hash
+from datetime import datetime
+from app import db
+
+rsvps = db.Table('rsvps',
+	db.Column('user_id', db.Integer, db.ForeignKey('users.id')),
+	db.Column('event_id', db.Integer, db.ForeignKey('events.id'))
+	)
+
+class User(db.Model):
+	"""this class to defines the user data model"""
+	__tablename__ = 'users'
+	id = db.Column(db.Integer, primary_key=True)
+	username = db.Column(db.String(64), unique=True)
+	email = db.Column(db.String(64), unique=True)
+	password_hash = db.Column(db.String(128))
+
+	@property
+	def password(self):
+		raise AttributeError('password is not a readable attribute')
+
+	@password.setter
+	def password(self, password):
+		self.password_hash = generate_password_hash(password)
+
+	def verify_password(self, password):
+		return check_password_hash(self.password_hash, password)
+
+	def __repr__(self):
+		return '<User %r>' % self.username
+
+class Events(db.Model):
+	"""This class represents the events table"""
+	__tablename__ = 'events'
+	id = db.Column(db.Integer, primary_key=True)
+	name = db.Column(db.String(64))
+	created_by = db.Column(db.String(64))
+	description = db.Column(db.Text)
+	category = db.Column(db.String(64))
+	location = db.Column(db.String(64))
+	event_date = db.Column(db.DateTime)
+	date_created = db.Column(db.DateTime, default=datetime.utcnow)
+	rsvps = db.relationship(User,
```

Comment:
Good implementation would like to know if it works and how you came about to this